platformio.ini

```diff
@@ -16,7 +16,7 @@
 
 [common]
 platform = espressif32
-platform_packages = framework-arduinoespressif32 @ https://github.com/espressif/arduino-esp32.git
+platform_packages = framework-arduinoespressif32 @ https://github.com/espressif/arduino-esp32.git#46d5afb17fb91965632dc5fef237117e1fe947fc
```

Comment:
I had to pin to this version to get the code to build now, it seems like arduino-esp32 has been updated in the last few days and broke things?

Reply:
Hmm, I'm not able to reproduce an error. I wonder if it was already fixed, can you try a `pio update` followed by a build? All still working for me:

```
$ pio update
Updating platformio/tool-scons                4.40100.2 @ ~4.40100.2             [Up-to-date]

Platform Manager
================
Platform espressif32
--------
Updating platformio/espressif32               3.1.1                              [Updating to 3.2.0]
Platform Manager: Installing platformio/espressif32 @ 3.2.0
Downloading  [####################################]  100%
Platform Manager: espressif32 @ 3.2.0 has been installed!
Tool Manager: toolchain-xtensa32 @ 2.50200.97 is already installed
Tool Manager: tool-esptoolpy @ 1.30000.201119 is already installed
Platform Manager: Removing espressif32 @ 3.1.1
Platform Manager: espressif32 @ 3.1.1 has been removed!
Updating platformio/toolchain-xtensa32        2.50200.97 @ ~2.50200.0            [Up-to-date]
Updating platformio/tool-esptoolpy            1.30000.201119 @ ~1.30000.0        [Up-to-date]
Updating platformio/tool-mkspiffs             2.230.0 @ ~2.230.0                 [Up-to-date]


Library Manager
===============
Library Storage: /Users/me/.platformio/lib
```

```
$ pio run -e m5stickc
[...]
================================================= [SUCCESS] Took 29.34 seconds =================================================

Environment    Status    Duration
-------------  --------  ------------
m5stickc       SUCCESS   00:00:29.342
```

Reply:
Still fails for me with:

`.pio/libdeps/m5stickc/WiFiManager/WiFiManager.cpp: In member function 'void WiFiManager::WiFiEvent(arduino_event_id_t, system_event_info_t)':
.pio/libdeps/m5stickc/WiFiManager/WiFiManager.cpp:3477:17: warning: comparison between 'enum arduino_event_id_t' and 'enum system_event_id_t' [-Wenum-compare]
     if(event == SYSTEM_EVENT_STA_DISCONNECTED){
                 ^
.pio/libdeps/m5stickc/WiFiManager/WiFiManager.cpp:3494:20: warning: comparison between 'enum arduino_event_id_t' and 'enum system_event_id_t' [-Wenum-compare]
   else if(event == SYSTEM_EVENT_SCAN_DONE){
                    ^
.pio/libdeps/m5stickc/WiFiManager/WiFiManager.cpp: In member function 'void WiFiManager::WiFi_autoReconnect()':
.pio/libdeps/m5stickc/WiFiManager/WiFiManager.cpp:3511:79: error: no matching function for call to 'WiFiClass::onEvent(std::_Bind_helper<false, void (WiFiManager::*)(arduino_event_id_t, system_event_info_t), WiFiManager*, const std::_Placeholder<1>&, const std::_Placeholder<2>&>::type)'
       wm_event_id = WiFi.onEvent(std::bind(&WiFiManager::WiFiEvent,this,_1,_2));
                                                                               ^
In file included from C:/Users/ukrnekm/.platformio/packages/framework-arduinoespressif32@src-537c58760dafe7fcc8a1d9bbcf00b6f6/libraries/WiFi/src/WiFiSTA.h:28:0,
                 from C:/Users/ukrnekm/.platformio/packages/framework-arduinoespressif32@src-537c58760dafe7fcc8a1d9bbcf00b6f6/libraries/WiFi/src/WiFi.h:32,
                 from .pio/libdeps/m5stickc/WiFiManager/WiFiManager.h:63,
                 from .pio/libdeps/m5stickc/WiFiManager/WiFiManager.cpp:13:
C:/Users/ukrnekm/.platformio/packages/framework-arduinoespressif32@src-537c58760dafe7fcc8a1d9bbcf00b6f6/libraries/WiFi/src/WiFiGeneric.h:145:21: note: candidate: wifi_event_id_t WiFiGenericClass::onEvent(WiFiEventCb, arduino_event_id_t)
     wifi_event_id_t onEvent(WiFiEventCb cbEvent, arduino_event_id_t event = ARDUINO_EVENT_MAX);
                     ^
C:/Users/ukrnekm/.platformio/packages/framework-arduinoespressif32@src-537c58760dafe7fcc8a1d9bbcf00b6f6/libraries/WiFi/src/WiFiGeneric.h:145:21: note:   no known conversion for argument 1 from 'std::_Bind_helper<false, void (WiFiManager::*)(arduino_event_id_t, system_event_info_t), WiFiManager*, const std::_Placeholder<1>&, const std::_Placeholder<2>&>::type {aka std::_Bind<std::_Mem_fn<void (WiFiManager::*)(arduino_event_id_t, system_event_info_t)>(WiFiManager*, std::_Placeholder<1>, std::_Placeholder<2>)>}' to 'WiFiEventCb {aka void (*)(arduino_event_id_t)}'
C:/Users/ukrnekm/.platformio/packages/framework-arduinoespressif32@src-537c58760dafe7fcc8a1d9bbcf00b6f6/libraries/WiFi/src/WiFiGeneric.h:146:21: note: candidate: wifi_event_id_t WiFiGenericClass::onEvent(WiFiEventFuncCb, arduino_event_id_t)
     wifi_event_id_t onEvent(WiFiEventFuncCb cbEvent, arduino_event_id_t event = ARDUINO_EVENT_MAX);
                     ^
C:/Users/ukrnekm/.platformio/packages/framework-arduinoespressif32@src-537c58760dafe7fcc8a1d9bbcf00b6f6/libraries/WiFi/src/WiFiGeneric.h:146:21: note:   no known conversion for argument 1 from 'std::_Bind_helper<false, void (WiFiManager::*)(arduino_event_id_t, system_event_info_t), WiFiManager*, const std::_Placeholder<1>&, const std::_Placeholder<2>&>::type {aka std::_Bind<std::_Mem_fn<void (WiFiManager::*)(arduino_event_id_t, system_event_info_t)>(WiFiManager*, std::_Placeholder<1>, std::_Placeholder<2>)>}' to 'WiFiEventFuncCb {aka std::function<void(arduino_event_id_t, arduino_event_info_t)>}'
C:/Users/ukrnekm/.platformio/packages/framework-arduinoespressif32@src-537c58760dafe7fcc8a1d9bbcf00b6f6/libraries/WiFi/src/WiFiGeneric.h:147:21: note: candidate: wifi_event_id_t WiFiGenericClass::onEvent(WiFiEventSysCb, arduino_event_id_t)
     wifi_event_id_t onEvent(WiFiEventSysCb cbEvent, arduino_event_id_t event = ARDUINO_EVENT_MAX);
                     ^
C:/Users/ukrnekm/.platformio/packages/framework-arduinoespressif32@src-537c58760dafe7fcc8a1d9bbcf00b6f6/libraries/WiFi/src/WiFiGeneric.h:147:21: note:   no known conversion for argument 1 from 'std::_Bind_helper<false, void (WiFiManager::*)(arduino_event_id_t, system_event_info_t), WiFiManager*, const std::_Placeholder<1>&, const std::_Placeholder<2>&>::type {aka std::_Bind<std::_Mem_fn<void (WiFiManager::*)(arduino_event_id_t, system_event_info_t)>(WiFiManager*, std::_Placeholder<1>, std::_Placeholder<2>)>}' to 'WiFiEventSysCb {aka void (*)(arduino_event_t*)}'
At global scope:
cc1plus.exe: warning: unrecognized command line option '-Wno-frame-address'
*** [.pio\build\m5stickc\lib8fb\WiFiManager\WiFiManager.cpp.o] Error 1
.pio/libdeps/m5stickc/LCBUrl/src/LCBUrl.cpp: In member function 'IPAddress LCBUrl::getIP(const char*)':
.pio/libdeps/m5stickc/LCBUrl/src/LCBUrl.cpp:634:53: error: cannot convert 'ip4_addr*' to 'esp_ip4_addr_t* {aka esp_ip4_addr*}' for argument '3' to 'esp_err_t mdns_query_a(const char*, uint32_t, esp_ip4_addr_t*)'
         esp_err_t err = mdns_query_a(hn, 2000, &addr);
                                                     ^
At global scope:
cc1plus.exe: warning: unrecognized command line option '-Wno-frame-address'
*** [.pio\build\m5stickc\liba87\LCBUrl\LCBUrl.cpp.o] Error 1
========================================================================= [FAILED] Took 21.44 seconds =========================================================================
Environment    Status    Duration
-------------  --------  ------------`

Caused by this mega commit: https://github.com/espressif/arduino-esp32/commit/5502879a5b25e5fff84a7058f448be481c0a1f73 (you can see e.g. the changes to mdns_query_a).  This was from a completely clean tree.

I'll remove it anyway and just apply locally for my build.